ourhexenv.py

```diff
@@ -171,25 +172,31 @@ def reset(self):
             pygame.display.flip()
 
     def step(self, action):
-        #Check if the action is within the valid range.
+    # Check if the action is within the valid range
         if action not in range(self.action_space.n):
             raise ValueError("Illegal move: Action is out of bounds.")
-        # Handle pie rule
+
+        # Handle pie rule (only Player 2 can use it in the first round)
         if action == self.board_size * self.board_size:
             if self.agent_selection == "player_1":
                 raise ValueError("Illegal move: Pie rule can only be used by Player 2.")
             if not self.is_pie_rule_usable:
                 raise ValueError("Illegal move: Pie rule can only be used once.")
+            if np.sum(self.board == 1) == 0:
+                raise ValueError("Illegal move: Pie rule cannot be used before Player 1's move.")
 
-            # Use pie rule, if a (row,col) was 1, make it 0 and make (col,row) 1
+            # Flip the board state 
             x, y = np.where(self.board == 1)
```

Comment:
I don't think this is correct. `np.where` gives you a mask the same size as the board. For example, let [[0, 1], [0, 0]] be the board. Where gives you back a mask of the same shape, so you get back [[0, 1], [0,0]]. I think you are looking for `np.argwhere` instead, which gives you the indexes.

```
locs = np.argwhere(self.board == 1)
row, col = locs[0]
# This location was previously player 1, (1) but we know want to mark it as player 2 (2)
self.board[row][col] = 2
```

I don't understand why on the code we are doing
```
self.board[row][col] = 0
self.board[col][row] = 2
```
Are we rotating the board 180 degrees?

Reply:
The pie rule is implemented using swap, like this over [here](https://www.lutanho.net/play/hex.html).

Reply:
Ahhh. When I read "Swap", I always thought it was a stone swap: 1 becomes 2. But based on the piece of code above and the animation it does a reflection as well (row, col, 0) -> (col, row, 2). That's fine - thank you for clarifying. Can you please add a comment as well?

Reply:
Wanted to clean up our discussion here based on your comments and #29.
The original swapping should work. Can we simply keep this?